src/state_management.cpp

```diff
@@ -57,6 +57,10 @@ namespace ara::sm {
         if (stateClient != nullptr) {
             internalState = stateClient->SmGetState();
         }
+        if (executionClient != nullptr) {
+            executionClient->ReportApplicationState(internalState);
+        }
```

Comment:
This shouldn't be here - this function is only for updating internal SM state...
but maybe it's good idea to change it - think about it.

Reply:
Discussed offline